### PR TITLE
[doc] Add introductions to ambient_elements in llvm_sparse_runtime.md

### PR DESCRIPTION
**Pull request by jim19930609**
Related issue = #

### CODE REVIEW

File: docs/design/llvm_sparse_runtime.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ Layout of a `dense` SNode:
 
 `pointer` is usually the go-to sparse SNode you should consider. It allocates memory only when a cell is actually activated, and recycles to a memory pool once it is deactivated. This saves the memory resource in large-scale grid computation. You can mentally view it as an `std::array<Cell*, N>`.
 
+Upon initialization, Taichi preallocates a chunk of memory space, namely `ambient_elements`, which is shared across all the inactive `dynamic` and `pointer` SNodes. Therefore dereferencing an inactive `pointer` SNode will generate the default value stored in `ambient_elements` (usually zero).
```
Comment (k-ye) [inline docs/design/llvm_sparse_runtime.md:53]: Is it just an inactive `pointer` or inactive sparse? E.g. `bitmasked`

File: docs/design/llvm_sparse_runtime.md [hunk 2]
```diff
@@ -123,6 +125,8 @@ The deactivation and the checking-for-active procedures are quite similar. We om
 
 Below shows the layout of a `dynamic` SNode. Logically speaking, `dynamic` SNode can be viewed as `std::vector<int32_t>`. However, `dynamic` is implemented as a singly linked list of *chunks*.
 
+Upon initialization, Taichi preallocates a chunk of memory space, namely `ambient_elements`, which is shared across all the inactive `pointer` and `dynamic` SNodes. Therefore dereferencing an inactive `dynamic` SNode will generate the default value stored in `ambient_elements` (usually zero).
```
Comment (k-ye) [inline docs/design/llvm_sparse_runtime.md:128]: Why twice?

File: docs/design/llvm_sparse_runtime.md (hunk 1 above)
Comment (jim19930609) [reply docs/design/llvm_sparse_runtime.md:53]: It's inactive sparse snode (pointer or dynamic), not actually bitmasked (bitmasked is like dense + mask)

File: docs/design/llvm_sparse_runtime.md (hunk 2 above)
Comment (jim19930609) [reply docs/design/llvm_sparse_runtime.md:128]: done, thanks!

File: docs/design/llvm_sparse_runtime.md (hunk 1 above)
Comment (jim19930609) [reply docs/design/llvm_sparse_runtime.md]: Rephrased to use "Sparse SNode"

**Review by k-ye — APPROVED** [review]
LGTM!

**Pull request merged**